crates/server/src/admin/service.rs

```diff
@@ -262,6 +268,43 @@ async fn list_applications_handler(
     }
 }
 
+async fn get_application_details_handler(
+    Path(app_id): Path<String>,
+    Extension(state): Extension<Arc<AdminState>>,
+) -> impl IntoResponse {
+    let app_id_result = match ApplicationId::from_str(&app_id) {
```

Comment:
PS: you can write this as this:

```suggestion
    let app_id_result = match app_id.parse() {
```

or if you really want to be explicit about the type definition (though it's inferred correctly from it's use in `get_application` below):

```suggestion
    let app_id_result = match app_id.parse::<ApplicationId>() {
```